packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1896,6 +2468,14 @@ class _RouteEntry {
 
   bool get willBePresent => currentState.index <= _RouteLifecycle.idle.index;
   bool get isPresent => currentState.index <= _RouteLifecycle.remove.index;
+  bool get suitableForAnnouncement {
+    return currentState.index <= _RouteLifecycle.removing.index &&
+      currentState.index >= _RouteLifecycle.push.index;
+  }
+  bool get canTransitition {
```

Comment:
What does "canTransition" mean? Why is this limited to lifecycles between remove and push?

Reply:
I will update the name, this field should return true for route that are suitable for transition animation. I limited the lifecycle between remove and push because add/adding is not suitable for transition animation because it has to wait for top most route to finish pushing animation